pyccel/codegen/printing/fcode.py

```diff
@@ -395,18 +386,14 @@ def _print_Program(self, expr):
 
             # ...
             sep = self._print(SeparatorComment(40))
-            funcs = ''
-            if expr.funcs:
-                for i in expr.funcs:
-                    funcs = ('{funcs}\n'
-                             '{sep}\n'
-                             '{f}\n'
-                             '{sep}\n').format(funcs=funcs, sep=sep, f=self._print(i))
 
-                funcs = 'contains\n{0}'.format(funcs)
+            if expr.funcs:
+                contains = 'contains'
```

Comment:
Yes, I think you're right. But I could not test it because functions are now moved to a separate module, aren't they? So maybe all this is all dead code now.

Reply:
Yes I think it should be dead code

Reply:
It looks like dead code on codacy too

Reply:
Should we remove the handling of functions in a program then? @saidctb what do you think?

Reply:
yes I think we should remove it

Reply:
OK then. I will remove `funcs` and `interfaces` from class `Program` in **_pyccel/ast/core.py_**, and change function `_print_Program` accordingly. Anything else that needs removing?

Reply:
It's completely unrelated but I noticed that there's a `_print_FromImport` function but no `FromImport` class so that wants removing I think

Reply:
Good catch! I'll remove that too!

And what about `classes` in a `Program`? Should they be removed too?

Reply:
Yes they should always be defined in a `Module` too